src/shogun/lib/DynamicArray.h

```diff
@@ -621,21 +774,13 @@ template <class T> class CDynamicArray :public CSGObject
 		virtual void init()
 		{
 			set_generic<T>();
+			T* head = m_array.data();
+			m_parameters->add_vector(
+			    &head, &num_elements, "array", "Memory for dynamic array.");
```

Comment:
i would use `m_array.size()` instead of `num_elements` for the size of vector, as you actually wanna have the same vector deserialized, right? meaning if you have `m_array.size() = 15` but `num_elements = 5`, you still wanna have the same sized `m_array` deserialized...so
```m_parameters->add_vector(&head, &(m_array.size()), ....```